src/udev/udev-worker.c

```diff
@@ -237,7 +237,7 @@ void udev_broadcast_result(sd_device_monitor *monitor, sd_device *dev, EventResu
         if (result != EVENT_RESULT_SUCCESS) {
                 (void) device_add_property(dev, "UDEV_WORKER_FAILED", "1");
 
-                switch (result) {
+                switch ((int)result) {
```

Comment:
instead of casting the value here, why not just cast the result of the addition below to the right enum type?

Reply:
So, type casting into the enum is not working and hence it is required to typecast into integer.

Reply:
So, type casting into the enum is not working and hence it is required to typecast into integer.

Reply:
why is it not working? what's the error you are seeing?

Reply:
compiler error message would be something like:
error: case value ‘1’ not in enumerated type ‘EventResult’ [-Werror=switch]